programs/scorumd/log_configurator.cpp

```diff
@@ -10,121 +10,196 @@
-            {
-                auto file_appender = fc::json::from_string(s).as<file_appender_args>();
+                    fc::console_appender::config config;
+                    config.level_colors.emplace_back(fc::console_appender::level_color(fc::log_level::debug, fc::console_appender::color::white));
+                    config.level_colors.emplace_back(fc::console_appender::level_color(fc::log_level::info, fc::console_appender::color::green));
+                    config.level_colors.emplace_back(fc::console_appender::level_color(fc::log_level::warn, fc::console_appender::color::brown));
+                    config.level_colors.emplace_back(fc::console_appender::level_color(fc::log_level::error, fc::console_appender::color::red));
+                    config.stream = stream;
 
-                fc::path file_name = file_appender.file;
+                    logging_config.appenders.push_back(fc::appender_config::create_config<fc::console_appender>(appender.appender, fc::variant(config)));
+
+                    found_logging_config = true;
+                    continue;
+                }
+                catch (fc::bad_cast_exception&)
+                {
+                }
 
-                if (file_name.is_relative())
+                //check if appender is a gelf appender
+                try
+                {
+                    //check if stream can be network endpoint
+                    std::string endpoint = appender.stream;
+                    if (endpoint.find(':') == std::string::npos)
+                    {
+                        endpoint += ":"; //add default port
+                        endpoint += boost::lexical_cast<std::string>(DEFAULT_GELF_APPENDER_PORT);
+                    }
+                    fc::ip::endpoint::from_string(endpoint); //if stream is not a network endpoint it will throw
+
+                    auto gelf_appender = fc::json::from_string(s).as<gelf_appender_args>();
+
+                    fc::gelf_appender::config config;
+                    config.endpoint = endpoint;
+                    config.host_name = gelf_appender.host_name;
+                    config.version = (fc::string)SCORUM_BLOCKCHAIN_VERSION;
+
+                    if (args.count("witness"))
+                    {
+                        try
+                        {
+                            const std::vector<std::string>& witnesses = args["witness"].as<std::vector<std::string>>();
+                            if (!witnesses.empty())
+                            {
+                                std::stringstream stream;
+                                stream << "witness";
+                                for (const std::string& s : witnesses)
+                                {
+                                    stream <<" "<< fc::json::from_string(s).as<std::string>();
+                                }
+                                config.additional_info = stream.str();
```

Comment:
Maybe api and plugins set would be useful like witness info?

Reply:
We can see all this additional info in logs. Witness is taken here just to clarify and identify immediately from which node the log string was sent without view whole log text.